packages/react-apollo-hooks/src/useQuery.ts

```diff
@@ -0,0 +1,202 @@
+import ApolloClient, {
+  OperationVariables,
+  ObservableQuery,
+  ApolloError,
+  WatchQueryOptions,
+} from 'apollo-client';
+import {ObservableQueryFields} from 'react-apollo';
+import {DocumentNode} from 'graphql-typed';
+import {useEffect, useMemo, useState} from 'react';
+
+import {AsyncQueryComponentType} from '@shopify/react-graphql';
+
+import {
+  QueryHookOptions,
+  QueryHookResult,
+  QueryDocumentResolving,
+} from './types';
+import {useApolloClient} from './ApolloContext';
+import useGraphQLDocument from './useGraphQLDocument';
+import {
+  getCachedObservableQuery,
+  invalidateCachedObservableQuery,
+} from './queryCache';
+import {compact, objToKey, actHack} from './utils';
+
+export default function useQuery<Data = any, Variables = OperationVariables>(
+  queryOrComponent: DocumentNode<Data, Variables>,
+  options?: QueryHookOptions<Data, Variables>,
+): QueryHookResult<Data, Variables>;
+
+export default function useQuery<
+  Data = any,
+  Variables = OperationVariables,
+  DeepPartial = {}
+>(
+  queryOrComponent: AsyncQueryComponentType<Data, Variables, DeepPartial>,
+  options?: QueryHookOptions<Data, Variables>,
+): QueryDocumentResolving<Data, Variables> | QueryHookResult<Data, Variables>;
+
+export default function useQuery<
+  Data = any,
+  Variables = OperationVariables,
+  DeepPartial = {}
+>(
+  queryOrComponent:
+    | DocumentNode<Data, Variables>
+    | AsyncQueryComponentType<Data, Variables, DeepPartial>,
+  options: QueryHookOptions<Data, Variables> = {},
+): QueryHookResult<Data, Variables> | QueryDocumentResolving<Data, Variables> {
+  const {
+    skip = false,
+    pollInterval,
+    notifyOnNetworkStatusChange = false,
+    client: overrideClient,
+    context,
+    variables,
+    fetchPolicy,
+    errorPolicy,
+  } = options;
+
+  const [responseId, setResponseId] = useState(0);
+  const client = useApolloClient(overrideClient);
+  const query = useGraphQLDocument(queryOrComponent);
+
+  const watchQueryOptions: WatchQueryOptions<Variables> = useMemo(
+    () =>
+      compact({
+        context,
+        errorPolicy,
+        fetchPolicy,
+        notifyOnNetworkStatusChange,
+        pollInterval,
+        query,
+        variables,
+      }),
+    [
+      query,
+      pollInterval,
+      notifyOnNetworkStatusChange,
+      context && objToKey(context),
+      variables && objToKey(variables),
+      fetchPolicy,
+      errorPolicy,
+    ],
+  );
+
+  const observableQuery = useMemo(
+    () => {
+      if (!query) {
+        return;
```

Comment:
I would return `null` here personally

Reply:
What's the rationale behind that?

Reply:
To get consistency in using either `null` or `undefined` to mean "not present yet"